rules/utils/is-promise-method-with-array.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+const METHODS = new Set(['all', 'allSettled', 'any', 'race']);
+
+const isPromiseMethodWithArray = node =>
+	node.callee.type === 'MemberExpression'
+	&& node.callee.object.type === 'Identifier'
+	&& node.callee.object.name === 'Promise'
+	&& node.callee.property.type === 'Identifier'
+	&& METHODS.has(node.callee.property.name)
+	&& node.arguments.length === 1
+	&& node.arguments[0].type === 'ArrayExpression';
```

Comment:
Can we switch to use [`isMethodCall`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/eb5af8bbd95f6bce3ad550c767bb63cec835719e/rules/ast/is-method-call.js#L26)?

```js
Promise[all]()
```

should not match.

Reply:
The proposed function doesn't seem to work with the bracket notation, so I've created a custom function.

Reply:
We don't need check `Promise["all"]`, no one write like this.

`Promise[all]` is not `Promise.all`, the it should be ignored.

Reply:
I misunderstood, sorry, I made the switch.